src/main/java/com/github/krzysiekjodlowski/ox/board/Board.java

```diff
@@ -0,0 +1,104 @@
+package com.github.krzysiekjodlowski.ox.board;
+
+import com.github.krzysiekjodlowski.ox.Event;
+import com.github.krzysiekjodlowski.ox.Move;
+import com.github.krzysiekjodlowski.ox.NumberLowerThanOneException;
+import com.github.krzysiekjodlowski.ox.Subscriber;
+import com.github.krzysiekjodlowski.ox.model.FieldNumber;
+import com.github.krzysiekjodlowski.ox.model.Symbol;
+
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * Store fields marked by players.
+ *
+ * @author Krzysztof Jodlowski
+ */
+public class Board implements Subscriber {
+  private final Map<FieldNumber, Symbol> fields = new HashMap<>();
+  private final int boardSideLength;
+  private final int boardCapacity;
+
+  public Board(final int boardSideLength) {
+    this.boardSideLength = boardSideLength;
+    this.boardCapacity = this.boardSideLength * this.boardSideLength;
+  }
+
+  /**
+   * Mark field when player makes move.
+   *
+   * @param event represents players move
+   */
+  @Override
+  public void handle(Event<?> event) {
+    Move playersMove = (Move) event;
+    this.fields.put(playersMove.getFieldNumber(), playersMove.getPlayersSymbol());
+  }
+
+  /**
+   * Used for checking field existence.
+   *
+   * @param fieldNumber field number
+   * @return true if field is already marked, false otherwise
+   */
+  public boolean containsField(FieldNumber fieldNumber) {
+    return this.fields.containsKey(fieldNumber);
+  }
+
+  int getBoardSideLength() {
+    return this.boardSideLength;
+  }
+
+  int getBoardCapacity() {
+    return this.boardCapacity;
+  }
+
+  int getBoardFulfillment() {
+    return this.fields.size();
+  }
+
+  Symbol getSymbol(FieldNumber fieldNumber) {
+    return this.fields.get(fieldNumber);
+  }
+
+  /**
+   * Creates nice String representation of a board game.
+   *
+   * @return String representation
+   */
+  @Override
+  public String toString() {
+    int maxFieldLength = Integer.toString(this.boardCapacity).length();
+    StringBuilder stringBuilder = new StringBuilder();
+
+    for (int i = 1; i <= this.boardCapacity; i++) {
+      FieldNumber currentFieldNumber = null;
+      try {
+        currentFieldNumber = FieldNumber.valueOf(i);
+      } catch (NumberLowerThanOneException e) {
+        e.printStackTrace();
```

Comment:
Not nice.

Reply:
I know, problem is that in this case, if nobody will change this class (ok it can be a weak explanation;) the iteration starts from one, always. I'll change it printing exception message.